ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -479,8 +480,10 @@ peerSelectionGovernor :: ( Alternative (STM m)
                       -> PeerSelectionActions peeraddr peerconn m
                       -> PeerSelectionPolicy  peeraddr m
                       -> PeerSelectionInterfaces peeraddr peerconn m
+                      -> STM m (Maybe LedgerPeerSnapshot)
+                      -> LedgerPeersConsensusInterface m
                       -> m Void
-peerSelectionGovernor tracer debugTracer countersTracer fuzzRng consensusMode actions policy interfaces =
+peerSelectionGovernor tracer debugTracer countersTracer fuzzRng consensusMode actions policy interfaces readBigLedgerSnapshot ledgerPeersCtx =
```

Comment:
You should not pass `LedgerPeersConsensusInterface` here. You should cascade the function you need through `PeerSelectionActions`, notice that PeerSelectionActions contains `readLedgerStateJudgement`

Reply:
The same for `STM m (Maybe LedgerPeerSnapshot)` put it inside `PeerSelectionInterfaces`

Reply:
Done